frontend/lib/resolution/scope-types.cpp

```diff
@@ -96,6 +96,13 @@ const Scope* Scope::moduleScope() const {
   return cur;
 }
 
+const Scope* Scope::parentModuleScope() const {
+  auto modScope = this->moduleScope();
+  if (this != modScope) return modScope;
```

Comment:
Hmm, can `import`s be used inside conditional blocks (they create scopes, don't they?) So then,
```Chapel
module M1 {
  module M2 {
    var x: int(32);
    if someCond {
      import super.x;
    }
  }
}
```
`someCond` is not a module, so `parentModuleScope` would return the scope corresponding to `M2`. So then it will find `x`. But this is not the case in Chapel (it reports `temp.chpl:6: error: Cannot find symbol 'x'`).

Reply:
For this snippet:
```Chapel
module M1 {
  var y: int(32);
  module M2 {
    var x: int(32);
    if someCond {
      import super.x;
      var target = x;
    }
  }
}
```
Your branch reports `target` to be `int(32)`, prod compiler reports `no variable x`.